### PR TITLE
Remove various single-scenario YAML parameters in our build scripts and shift implementation responsibility to those scenarios

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Uses Arcade defaults (defaults to false/empty string) to remove richCodeNavigation property.

Inline CodeQL injection to the CodeQL pipeline

Use Arcade's Source-Build jobs now that we aren't using any features that require the steps support.

CodeQL run at https://dev.azure.com/dnceng/internal/_build/results?buildId=2411429&view=results
Official Build at https://dev.azure.com/dnceng/internal/_build/results?buildId=2411430

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -1878,36 +1878,29 @@ extends:
...
+        # - Linux-x64 portable (used for dependency flow and downstream PR verification)
+        # - Banana.24 - Non-existent RID to ensure we don't break RIDs we don't know about.
+        #
+        # Running all of these everywhere is wasteful. Run Banana.24 and CentOS.8 in rolling CI,
+        # Run Linux-x64 in PR.
+        - template: /eng/common/templates/jobs/source-build.yml
+          parameters:
+            platforms:
+              - name: CentOS8
+                targetRid: centos.8-x64
```
Comment (hoyosjs) [inline eng/pipelines/runtime.yml:1899]: Isn't CentOS EOL soon?

File: src/installer/prepare-artifacts.proj [hunk 2]
```diff
@@ -145,10 +145,6 @@
         <RelativeBlobPath>$(InstallersRelativePath)workloads/%(Filename)%(Extension)</RelativeBlobPath>
         <PublishFlatContainer>true</PublishFlatContainer>
       </ItemsToPush>
-
-      <!-- Source build intermediated packages will be pushed and signed by the sourcebuild leg. -->
```
Comment (hoyosjs) [inline src/installer/prepare-artifacts.proj:149]: How come this changed

**Review by hoyosjs — APPROVED** [review]
Looks reasonable if official build passes

**Review by directhex — APPROVED** [review]

File: eng/pipelines/common/global-build-job.yml [hunk 3]
```diff
@@ -154,10 +145,7 @@ jobs:
 
     - checkout: self
       clean: true
-      # If running in source build mode, a git stash will be used for the inner clone. Avoid setting a fetch depth,
-      # as a stash of a shallow cloned repo is not currently supported.
-      ${{ if ne(parameters.isSourceBuild, true) }}:
-        fetchDepth: $(checkoutFetchDepth)
+      fetchDepth: $(checkoutFetchDepth)
```
Comment (hoyosjs) [inline eng/pipelines/common/global-build-job.yml:148]: Who sets this for SB?

File: src/installer/prepare-artifacts.proj (hunk 2 above)
Comment (jkoritzinsky) [reply src/installer/prepare-artifacts.proj:149]: Now that we aren't pushing a copy of the source-build artifacts into the IntermediateArtifacts artifact, we can remove this code that filters it out.

File: eng/pipelines/common/global-build-job.yml (hunk 3 above)
Comment (jkoritzinsky) [reply eng/pipelines/common/global-build-job.yml:148]: SourceBuild has never had this set (due to the comment above).

File: eng/pipelines/runtime.yml (hunk 1 above)
Comment (jkoritzinsky) [reply eng/pipelines/runtime.yml:1899]: Yes, but I wanted to preserve existing behavior in this PR to limit possible issues.

**Pull request merged**